src/v/redpanda/admin/api-doc/debug.json

```diff
@@ -786,6 +786,40 @@
                     }
                 }
             ]
+        },
+        {
+            "path": "/v1/debug/offset_for_leader_epoch/{topic}/{partition}/{epoch}",
```

Comment:
nit: maybe we could move it to be part of partition resource as an `offset_for_leader_epoch` is partition feature. Something like: 
`GET /v1/debug/partitions/{namespace}/{topic}/{partition}/offet_for_leader_epoch?epoch=<epoch>`

Reply:
done.